### DISCUSSION TITLE
MilvusException: <MilvusException: (code=2, message=Fail connecting to server on 10.159.34.189:19530, illegal connection params or server unavailable)>

### DISCUSSION THREAD

**Discussion by mashaonian** [milvus-io/milvus#36374] [Announcement]
docker compose部署的stanlone版的milvus，pymilvus只能通过localhost连接，不能通过宿主机的具体ip连接吗？我通过localhost能够连接成功，但是使用宿主机的具体ip却出现了服务不存在

**Comment by yhmo** [comment]
正常都是可以的，比如我机器本地局域网ip是192.168.240.1，我用connections.connect(host="192.168.240.1", port=19530) 就能正常连接上。

**Reply by mashaonian** [reply]
> 正常都是可以的，比如我机器本地局域网ip是192.168.240.1，我用connections.connect(host="192.168.240.1", port=19530) 就能正常连接上。

请问你也是用wsl下的ubuntu子系统，并在ubuntu中使用docker compose部署的milvus吗，你是否能够telnet your_ip 19530

**Reply by yhmo** [reply]
我的不是wsl ubuntu，而是native的ubuntu，是用docker compose部署，telnet 19530没问题。你的问题只跟wsl相关，和milvus没关系。

**Reply by mashaonian** [reply]
> 我的不是wsl ubuntu，而是native的ubuntu，是用docker compose部署，telnet 19530没问题。你的问题只跟wsl相关，和milvus没关系。

那可能是我的wsl有问题，并且我的telnet 19530也是无法连接，你知道怎么解决吗？

**Reply by yhmo** [reply]
你可以在网上搜索相关的问题，比如下面这些可能跟你的问题相似：
https://forums.docker.com/t/wsl-2-and-port-forwarding/94891
https://stackoverflow.com/questions/69838995/wsl2-docker-ports-are-not-openend-on-host

**Reply by mashaonian** [reply]
> 你可以在网上搜索相关的问题，比如下面这些可能跟你的问题相似： https://forums.docker.com/t/wsl-2-and-port-forwarding/94891 https://stackoverflow.com/questions/69838995/wsl2-docker-ports-are-not-openend-on-host

真是太感谢你为我找到错误的方向，现在终于把这个困扰我很久的问题解决了。
